fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneLockFailureTest.java

```diff
@@ -0,0 +1,261 @@
+            grabLockExternally(SIMPLE_INDEX, context);
+            context.commit();
+        }
+
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, SIMPLE_INDEX);
+            // This fails since the merge tries to take a lock
+            Assertions.assertThrows(FDBExceptions.FDBStoreLockTakenException.class, () ->
+                    LuceneIndexTestUtils.mergeSegments(recordStore, SIMPLE_INDEX));
+        }
+    }
+
+    @Test
+    void testRebalance() throws IOException {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+            // partition size is 10
+            for (int i = 0; i < 50; i++) {
+                recordStore.saveRecord(createComplexDocument(6666L + i, ENGINEER_JOKE, 1, Instant.now().toEpochMilli()));
+            }
+            context.commit();
+        }
+
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+            grabLockExternallyForPartition(COMPLEX_PARTITIONED, context, 1);
+            context.commit();
+        }
+
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+            // This fails since the repartition tries to take a lock
+            // The exception here is the RecordCore wrapper around the Lucene exception
+            Assertions.assertThrows(RecordCoreException.class, () ->
+                    LuceneIndexTestUtils.rebalancePartitions(recordStore, COMPLEX_PARTITIONED));
+        }
+    }
+
+    @Test
+    void testLockTwice() throws IOException {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, SIMPLE_INDEX);
+            grabLockExternally(SIMPLE_INDEX, context);
+            context.commit();
+        }
+
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, SIMPLE_INDEX);
+            // The exception here is the Lucene IOException
+            Assertions.assertThrows(LockObtainFailedException.class, () ->
+                    grabLockExternally(SIMPLE_INDEX, context));
+            context.commit();
+        }
+    }
+
+    private void grabLockExternally(final Index index, final FDBRecordContext context) throws IOException {
+        final FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions());
+        directory.obtainLock(IndexWriter.WRITE_LOCK_NAME);
+    }
+
+    private void grabLockExternallyForPartition(final Index index, final FDBRecordContext context, int partition) throws IOException {
+        final Subspace partitionSubspace = recordStore.indexSubspace(index).subspace(Tuple.from(partition, LucenePartitioner.PARTITION_DATA_SUBSPACE).add(0));
```

Comment:
The partition in this case is `0`, you are putting the group in the subspace.
```suggestion
    private void grabLockExternallyForPartition(final Index index, final FDBRecordContext context, int group) throws IOException {
        final Subspace partitionSubspace = recordStore.indexSubspace(index).subspace(Tuple.from(group, LucenePartitioner.PARTITION_DATA_SUBSPACE).add(0));
```

Reply:
Yes. Done. Good catch.